modality-ros-hook/src/message_processor.rs

```diff
@@ -177,48 +440,206 @@ impl MessageProcessor {
+                        self.interned_attr_key(&format!(
+                            "timeline.ros.service.client_guid.{}",
+                            hex::encode(service_event.client_guid)
+                        ))
+                        .await?,
+                        AttrVal::Bool(true),
+                    ),
+                    (
+                        self.interned_attr_key(&format!(
+                            "timeline.ros.service.{}",
+                            normalized_service_name
+                        ))
+                        .await?,
+                        AttrVal::Bool(true),
+                    ),
+                ];
+                self.client.timeline_metadata(tl_attrs).await?;
+            }
+            ServiceEventType::TakeRequest => (),
+            ServiceEventType::SendResponse => (),
+            ServiceEventType::TakeResponse => (),
         };
 
-        // If we have a cached id, just make this the active timeline and be done.
-        if let Some(tl_cache_val) = self.timeline_cache.get(&key) {
-            if self.currently_open_timeline != Some(tl_cache_val.id) {
-                self.client.open_timeline(tl_cache_val.id).await?;
-                self.currently_open_timeline = Some(tl_cache_val.id);
+        let mut event_attrs = vec![
+            (
+                self.interned_attr_key("event.name").await?,
+                AttrVal::String(Cow::Owned(normalized_service_name.clone())),
+            ),
+            (
+                self.interned_attr_key("event.ros.service.name").await?,
+                AttrVal::String(Cow::Owned(service_event.service_name.clone())),
+            ),
+            (
+                self.interned_attr_key("event.ros.service.event").await?,
+                AttrVal::String(Cow::Owned(service_event.event_type.to_string())),
+            ),
+        ];
+
+        if let Some(ts) = service_event.timestamp {
+            event_attrs.push((
+                self.interned_attr_key("event.timestamp").await?,
+                AttrVal::Timestamp(ts),
+            ));
+        }
+
+        match service_event.event_type {
+            ServiceEventType::SendRequest | ServiceEventType::TakeResponse => {
+                event_attrs.push((
+                    self.interned_attr_key("event.ros.service.sequence_id")
+                        .await?,
+                    AttrVal::Integer(service_event.sequence_id),
+                ));
+                event_attrs.push((
+                    self.interned_attr_key("event.ros.service.client_guid")
+                        .await?,
+                    AttrVal::String(Cow::Owned(hex::encode(service_event.client_guid))),
+                ));
+            }
+            ServiceEventType::TakeRequest | ServiceEventType::SendResponse => {
+                event_attrs.push((
+                    self.interned_attr_key(&format!(
+                        "event.interaction.remote_timeline.ros.service.client_guid.{}",
```

Comment:
I don't have a good sense as to what should live under the `event.internal...` prefix, but maybe this should?